File: resources/exercises/sort/basic/comb/AlgCombSort11Entity.c
//RemoteSort

/* BEGIN TEMPLATE */
void combSort11() {
	/* BEGIN SOLUTION */
	int gap = getValueCount();
	int swapped;
	do {
		if (gap>1) {
			gap /= 1.3;
			if (gap == 10 || gap == 9)
				gap = 11;
		}
		swapped = 0;
		int i;
		for (i=0; i+gap<getValueCount(); i++)
			if (!isSmaller(i,i+gap)) {
				swap(i,i+gap);
				swapped =1;
			}
	} while (gap>1 || swapped);
	/* END SOLUTION */
}
/* END TEMPLATE */


void run(){
	combSort11();
}

File: resources/exercises/sort/basic/comb/AlgCombSort11Entity.py
# BEGIN SOLUTION 
gap = getValueCount()
swapped = True
while gap > 1 or swapped:
	if gap > 1:
		gap = int( gap / 1.3)
	if gap == 10 or gap == 9:
		gap = 11
	swapped = False
	i = 0
	while i+gap <  getValueCount():
		if not isSmaller(i,i+gap):
			swap(i, i+gap)
			swapped = True
		i += 1
# END SOLUTION
